core/modules/update/update.install

```diff
@@ -239,6 +239,31 @@ function update_update_1004() {
   $config->save();
 }
 
+/**
+ * Check installer.settings.json and fix it or remove it if needed.
```

Comment:
It took me a while reading the back issues to figure why we needed this function (about Installer, but in Update??), so I suspect others may benefit from a little bit more explanation here without needing to git blame in the future.

Reply:
Thanks, @docwilmot. I had originally added this to `installer.intall`, but @indigoxela rightly pointed out that the faulty installer.settings.json is created even if `installer` is not enabled, meaning that the deletion of the faulty file must happen outside `installer.install` (otherwise, the faulty config file will not get fixed or removed if `installer` is uninstalled). 

The culprit for creating this faulty and unnecessary file is an ealier update hook inside `update.install`. Therefore we both thought that creating this update hook within `update.install` was the right way to process. 

I can add some explanations for this in the update hook. However I did not want to confuse or overwhelm the administrator running update with information in the doc block (which is shown when you run update.php). Do you suggest adding as inline comments within the function?

Reply:
I agree with everything you folks did actually, it just took me a while to get it, so wanted to ensure we help the next person to come across this function. And yes, I think some inline would be great.